.github/workflows/Cpp-lint-check.yaml

```diff
@@ -0,0 +1,29 @@
+# GitHub Action to run cpplint recursively on all pushes and pull requests
+# https://github.com/cpplint/GitHub-Action-for-cpplint
+
+on:
+  push:
+    paths: "src/**"
+  pull_request:
+    branches:
+      - "*"
+
+name: Cpp-lint-check
+
+jobs:
+  cpplint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v2
+      - run: pip install cpplint
+      - run: cpplint --exclude="src/RcppExports.cpp" src/*.cpp
+      - run: cpplint --exclude="inst/include/finalsize.h" inst/include/*.h
+
+  cppcheck:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: sudo apt-get install cppcheck
+      - run: cppcheck --std=c++14 -i src/RcppExports.cpp --enable=style --error-exitcode=1 src
+      - run: cppcheck --std=c++14 --language=c++ --enable=style --error-exitcode=1 inst/include/*.h
```

Comment:
I think this could be safely changed to the C++17 standard (with corresponding changes in the DESCRIPTION and Makevars). After some checking, neither {finalsize} nor {epidemics} are affected by this, most likely because the codebases are small - Cppcheck doesn't seem to complain for older (C++11) or more recent standards (C++17, C++20), and that might just be because it's quite a permissive checker.

Reply:
Fixed in https://github.com/epiverse-trace/finalsize/pull/161/commits/612cecd8a6053930ca2c1688c356307793fa02c0